PC/winreg.c

```diff
@@ -749,21 +764,22 @@ Reg2Py(BYTE *retDataBuf, DWORD retDataSize, DWORD typ)
                 }
                 for (index = 0; index < s; index++)
                 {
-                    size_t len = wcslen(str[index]);
-                    if (len > INT_MAX) {
+                    size_t slen = wcsnlen(str[index], len);
```

Comment:
This might read past the end of `data`. For example, if the buffer is `L"spam\x00eggs"`, then `len` is 9 and `wcsnlen(str[1], 9)` will either include 5 extraneous characters or possibly even crash on an access violation.  Adjust for the remaining length using pointer arithmetic.

Reply:
I might be missing something, but I don't think that would happen: `len -= slen;` is performed on each iteration.

Reply:
Sorry, I overlooked the decrement at the end. But it should be `len -= slen + 1` because `slen` doesn't include the null.